[core][docs] Lint some top level core docs (#49703)
ray-project/ray: doc/source/ray-core/configure.rst

The following options specify the range of ports used by worker processes across machines. All ports in the range should be open.

- ``--min-worker-port``: Minimum port number for the worker to bind to. Default: 10002.
- ``--max-worker-port``: Maximum port number for the worker to bind to. Default: 19999.

Port numbers are how Ray differentiates input and output to and from multiple workers on a single node. Each worker takes input and gives output on a single port number. Thus, for example, by default, there is a maximum of 10,000 workers on each node, irrespective of number of CPUs.

> In general, it's recommended to give Ray a wide range of possible worker ports, in case any of those ports happen to be in use by some other program on your machine. However, when debugging it's useful to explicitly specify a short list of worker ports such as ``--worker-port-list=10000,10001,10002,10003,10004``

Reviewer: ```suggestion
In general, it's recommended to give Ray a wide range of possible worker ports, in case any of those ports happen to be in use by some other program on your machine. However, when debugging, it's useful to explicitly specify a short list of worker ports such as ``--worker-port-list=10000,10001,10002,10003,10004``
```

> Note that this limits the number of workers, just like specifying a narrow range.

Reviewer: ```suggestion
Note that this practice limits the number of workers, just like specifying a narrow range.
```

relatively smaller for large workloads.
The exact overhead depends on the nature of your workload.

Java applications
-----------------

> .. important:: For the multi-node setting, you must first run ``ray start`` on the command line to start the Ray cluster services on the machine before ``Ray.init()`` in Java to connect to the cluster services. On a single machine, you can run ``Ray.init()`` without ``ray start``. It both starts the Ray cluster services and connects to them.

Reviewer: ```suggestion
.. important:: For the multi-node setting, you must first run ``ray start`` on the command line to start the Ray cluster services on the machine before ``Ray.init()`` in Java to connect to the cluster services. On a single machine, you can run ``ray.init()`` without ``ray start``. It both starts the Ray cluster services and connects to them.
```

.. _code_search_path:

Code search path
~~~~~~~~~~~~~~~~

> If you want to run a Java application in a multi-node cluster, you must specify the code search path in your driver. The code search path is to tell Ray where to load jars when starting Java workers. You must distribute your jar files to the same paths on all nodes of the Ray cluster before running your code.

Reviewer: ```suggestion
If you want to run a Java application in a multi-node cluster, you must specify the code search path in your driver. The code search path tells Ray where to load jars when starting Java workers. You must distribute your jar files to the same paths on all nodes of the Ray cluster before running your code.
```

      -Dray.job.code-search-path=/path/to/jars/ \
      <classname> <args>

> The ``/path/to/jars/`` here points to a directory which contains jars. Workers load all jars in the directory. You can also provide multiple directories for this parameter.

Reviewer: ```suggestion
The ``/path/to/jars/`` points to a directory, which contains jars. Workers load all jars in the directory. You can also provide multiple directories for this parameter.
```

See ``ray.job.code-search-path`` under :ref:`Driver Options <java-driver-options>` for more information.

.. note:: Currently there's no way to configure Ray when running a Java application in single machine mode. If you need to configure Ray, run ``ray start`` to start the Ray cluster first.

.. _java-driver-options:

Driver options
~~~~~~~~~~~~~~

> There is a limited set of options for Java drivers. they're not for configuring the Ray cluster, but only for configuring the driver.

Reviewer: ```suggestion
There's a limited set of options for Java drivers. They're not for configuring the Ray cluster, but only for configuring the driver.
```

- ``ray.address``

  - The cluster address if the driver connects to an existing Ray cluster. If it's empty, Ray creates a new Ray cluster.
  - Type: ``String``
  - Default: empty string.

- ``ray.job.code-search-path``

>   - The paths for Java workers to load code from. Currently, Ray only supports directories. You can specify one or more directories split by a ``:``. You don't need to configure code search path if you run a Java application in single machine mode or local mode. Code search path is also used for loading Python code if it's specified. This is required for :ref:`cross_language`. If you specify a code search path, you can only run Python remote functions which you can find in the code search path.

Reviewer: ```suggestion
  - The paths for Java workers to load code from. Currently, Ray only supports directories. You can specify one or more directories split by a ``:``. You don't need to configure code search path if you run a Java application in single machine mode or local mode. Ray uses code search path for loading Python code, if specified. This parameter is required for :ref:`cross_language`. If you specify a code search path, you can only run Python remote functions which you can find in the code search path.
```

  - Type: ``String``
  - Default: empty string.
  - Example: ``/path/to/jars1:/path/to/jars2:/path/to/pys1:/path/to/pys2``

- ``ray.job.namespace``

>   - The namespace of this job. It's used for isolation between jobs. Jobs in different namespaces can't access each other. If it's not specified, Ray uses a randomized value.

Reviewer: ```suggestion
  - The namespace of this job. Ray uses it for isolation between jobs. Jobs in different namespaces can't access each other. If it's not specified, Ray uses a randomized value.
```
